src/main/java/net/minecraftforge/fluids/FluidRegistry.java

```diff
@@ -356,6 +356,21 @@ public static String getDefaultFluidName(Fluid key)
         return name;
     }
 
+    @Nullable
+    public static String getModId(@Nullable FluidStack fluidStack)
+    {
+        if (fluidStack != null)
```

Comment:
I was suggesting that `Item::getModId` look

```java
public Optional<String> getModId(@Nonnull ItemStack stack)
{
    return Optional.ofNullable(getRegistryName())
        .map(ResourceLocation::getResourceDomain);
}
```

and if other new methods were refactored like this

```java
public static Optional<String> FluidRegistry.getModId(@Nonnull FluidStack stack)
{
    Optional.ofNullable(getDefaultFluidName(stack.getFluid))
      .map(ResourceLocation::new) // Maybe string splitting instead?
      .map(ResourceLocation::getResourceDomain);
}
@Override public Optional<String> UniversalBucket::getModID(@Nonnull ItemStack stack) {
    Optional.ofNullable(getFluid(stack))
      .flatMap(FluidRegistry::getModID)
      .map(Optional::of).orElseGet(() -> super.getModID(stack)) // I admit this is a bit ew
}
```

etc.